allure-kotlin-migration-guide.md

```diff
@@ -0,0 +1,94 @@
+    1. Change imports for Allure Junit4 annotations `DisplayName`, `Tag`, `Tags` from `io.qameta.allure.android.annotations.*` -> `io.qameta.allure.kotlin.junit4.*`
+    1. Change import for `SeverityLevel` from `io.qameta.allure.android.SeverityLevel` -> `io.qameta.allure.kotlin.SeverityLevel`
+1. Migrate package `io`
+    1. Change all imports from `io.qameta.allure.android.io.*` -> `io.qameta.allure.kotlin.*`
+    1. In case you have custom `AllureResultsWriter` adjust it as the API has changed a bit
+    1. `AllureResultsReader` & `FileSystemResultsReader` have been removed as they are not used anymore
+1. Migrate package `listener` - change all imports from `io.qameta.allure.android.listener.*` -> `io.qameta.allure.kotlin.listener.*`
+1. Migrate package `serialization` - it was used internally and is no longer needed.
+1. Migrate package `utils` - it was used to parse the annotations to allure results. It shouldn't be accesed by consumer code. If you were using any of those methods take a look at ` io.qameta.allure.kotlin.util.ResultsUtils` for replacement (as the implementation of parsing annotations has changed).
+ 
+
+### Step 3 - `allure-espresso` migration
+1. Replace dependency
+   ```gradle
+   dependencies {
+       //REMOVE allure-android dependency
+       androidTestImplementation "io.qameta.allure:allure-espresso:$ALLURE_ANDROID_VERSION"
+  
+       //ADD allure-kotlin dependency
+       androidTestImplementation "io.qameta.allure:allure-kotlin-android:$ALLURE_KOTLIN_VERSION"
+   }
+1. Migrate runners (don't forget migrating `testInstrumentationRunner` gradle configuration)
+    1. Change import for `AllureAndroidRunner` from `io.qameta.allure.espresso.AllureAndroidRunner` -> `io.qameta.allure.android.runners.AllureAndroidJUnitRunner`
+    1. Change import for `MultiDexAllureAndroidRunner` from `io.qameta.allure.espresso.MultiDexAllureAndroidRunner` -> `io.qameta.allure.android.runners.MultiDexAllureAndroidJUnitRunner` 
+1. Replace `AllureAndroidListener` usages with `io.qameta.allure.kotlin.junit4.AllureJunit4`. In case you want to have external storage permissions automatically granted as well (default behaviour of `AllureAndroidListener`) attach addititional test listener `io.qameta.allure.android.listeners.ExternalStoragePermissionsListener`
+1. Migrate test rules
+    1. Replace `FailshotRule` with `io.qameta.allure.android.rules.ScreenshotRule` that includes configuration to take a screenshot after sucessfull tests.
+    1. Replace `LogcatClearRule` & `LogcatDumpRule` with `io.qameta.allure.android.rules.LogcatRule` that merges behaviour of those 2 rules.
+    1. Change import for `WindowHierarchyRule` to `io.qameta.allure.android.rules.WindowHierarchyRule`
+1. Change import for `AllureAndroidLifecycle` from `io.qameta.allure.espresso.io.qameta.allure.espresso` -> `io.qameta.allure.android.AllureAndroidLifecycle`
```

Comment:
Could you also add a few words about `AllureAndroidLifecycle.addAttachment`, cause signature of the method has been changed.